### ISSUE TITLE
hoverReply: type mismatch

### ISSUE THREAD

**Issue by Shane-XB-Qian** [yegappan/lsp#6]
![image](https://user-images.githubusercontent.com/13550529/109323565-8fe87f00-788e-11eb-99be-fe09aeedf6bb.png)

**Comment by Shane-XB-Qian** [comment]
if hover failed (e1012), then ProcessMessages failed too, no that/such msg.
// trying to fix with pr #4

**Comment by yegappan** [comment]
Thanks for reporting the problem. I have pushed a commit that should address this error. Can you try the latest plugin?

**Comment by Shane-XB-Qian** [comment]
um, looks no, pls check 2ed commit eb926526b0e3b794c68190373229d7ff0c3f8097 within pr #4
the init problem is here:
```vim
    elseif reply.result.contents->has_key('value')
      # MarkedString
      hoverText = reply.result.contents.value->split("\n")
    else
```
missed `split`.

**Comment by yegappan** [comment]
Got it. Can you pull a separate PR with this change, I will merge it?

**Comment by Shane-XB-Qian** [comment]
perhaps you can reset this/latest check-in, 
then just merge pr #4 first, and integrated your other changes in `lspserver.vim`.
looks would be easy if so, otherwise, current code had conflict, it maybe a bit complex to solve.
// but wait, let me add some description as you said in https://github.com/yegappan/lsp/pull/4#issuecomment-786767500, then you do above?

**Comment by Shane-XB-Qian** [comment]
description (doc help) added for pr #4 now.
// that's type mismatch init happened at hover function too.

**Comment by Shane-XB-Qian** [comment]
solved code/commit conflict with your last commit, now should be clear and can be direct to merge pr #4 if ok. thx.

// partially code change which related to this issue had been covered by your/this last commit. so did not include it now.
// and i saw your last commit solved the type mis-match for 'lspFormat' cmd as well / now, which was a knew issue too. thx.

**Comment by Shane-XB-Qian** [comment]
this can be closed, i saw you check in another commit 816e8ac9e5ad7c6fbdbd6804bb5926f954630852 had been covered all.
